arch/x86/core/ia32/float.c

```diff
@@ -173,17 +173,22 @@ void z_float_enable(struct k_thread *thread, unsigned int options)
 
 	imask = irq_lock();
 
-	/* Indicate thread requires floating point context saving */
-
-	thread->base.user_options |= (uint8_t)options;
-
 	/*
 	 * The current thread might not allow FP instructions, so clear CR0[TS]
 	 * so we can use them. (CR0[TS] gets restored later on, if necessary.)
 	 */
 
 	__asm__ volatile("clts\n\t");
 
+	if (!thread) {
+		irq_unlock(imask);
+		return;
+	}
```

Comment:
This check is not making sense. If you need it, this should be done the `irq_lock()` to avoid the locking/unlocking.

The other thing is, there is any chance this parameter be NULL ? Usually Zephyr does not check function parameters in internal functions unless it is necessary.

Reply:
Agree that can be done without locking, update the commit, thanks.
The case we met is, _FpNotAvailableExcHandler is triggered at early boot stage before _current is valid; and there defined  syscall: float_enable which will call this, add this check will make it safe.  please check whether this reasonable, thanks.